### PR TITLE
 Fixes #22020 - Host group play roles using REX provider

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

File: app/controllers/foreman_ansible/concerns/hosts_controller_extensions.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def multiple_play_roles
           @composer.trigger
           redirect_to job_invocation_path(@composer.job_invocation)
         else
-          #raise ::Foreman::Exception.new(N_("Could not run Ansible roles for %s") % @host)
+          raise ::Foreman::Exception.new(N_("Could not run Ansible roles %s") % @composer.errors)
```
Comment (ares) [inline app/controllers/foreman_ansible/concerns/hosts_controller_extensions.rb]: this is quite specific exception, should we create new child class of Foreman::Exception for it? something like AnsibleRolePlayError

File: app/controllers/foreman_ansible/concerns/hostgroups_controller_extensions.rb [hunk 2]
```diff
@@ -18,6 +19,13 @@ def play_roles
 
       private
 
+      def check_hostgroup
+        return unless @hostgroup.hosts.empty?
```
Comment (ares) [inline app/controllers/foreman_ansible/concerns/hostgroups_controller_extensions.rb]: this is another place to be changed if we decide to run ansible on hosts assigned to children hostgroups as well but good for now

**Comment by ares** [comment]
looking at last version, I'm not sure failures from http://ci.theforeman.org/job/test_plugin_matrix/4613/ were fixed, did you trigger another jenkins run? otherwise LGTM once we have green run on postgres/mysql

**Comment by ares** [comment]
@dLobatog both seems to be merged now

**Comment by dLobatog** [comment]
Yup, all green but sqlite (rex problem) - ci.theforeman.org/job/test_plugin_matrix/4630/ - merging

**Pull request merged**